File: packages/backend/src/features/game/game.controller.ts
import { Body, Controller, Get, Post, Put } from '@nestjs/common';
import { UUID } from 'node:crypto';
import { CreateGameDto } from '~/src/features/game/dto/create-game.dto';
import { GameDto } from '~/src/features/game/dto/game.dto';
import { GameMapper } from '~/src/features/game/game.mapper';
import { GameService } from '~/src/features/game/game.service';

@Controller('games')
export class GameController {
  private readonly gameService: GameService;

  constructor(gameService: GameService) {
    this.gameService = gameService;
  }

  @Get()
  async getGames(): Promise<GameDto[]> {
    const gameModels = await this.gameService.getGames();
    return GameMapper.toDtoList(gameModels);
  }

  @Post()
  async createGame(
    @Body() createGameDto: CreateGameDto,
  ): Promise<{ id: UUID }> {
    const gameModel = await this.gameService.createGame(createGameDto);
    return { id: GameMapper.toDto(gameModel).id };
  }

  @Put()
  async updateGame() {
    throw new Error('Not implemented');
  }
}

File: packages/backend/src/main/kotlin/quizzem/backend/features/game/api/GameController.kt
package quizzem.backend.features.game.api

import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import quizzem.backend.features.game.api.dto.CreateGameDto
import quizzem.backend.features.game.service.GameService

@RestController
@RequestMapping("/game")
class GameController(private val gameService: GameService) {

    @PostMapping
    fun createGame(@RequestBody() dto: CreateGameDto) {
        return gameService.createGame(dto)
    }
}